### ISSUE TITLE
Cors crate fails to compile: expected struct `actix_web::BaseHttpResponse`, found struct `actix_web::HttpResponse`

### ISSUE THREAD

**Issue by tobisinghania** [actix/actix-extras#176]
## Current Behavior
When adding `actix-cors = "0.6.0-beta.1"` to a project, the compilation fails with
```
   ....
   Compiling actix-cors v0.6.0-beta.1
error[E0053]: method `error_response` has an incompatible type for trait
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\actix-cors-0.6.0-beta.1\src\error.rs:53:5
   |
53 |     fn error_response(&self) -> HttpResponse {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `actix_web::BaseHttpResponse`, found struct `actix_web::HttpResponse`
   |
   = note: expected fn pointer `fn(&error::CorsError) -> actix_web::BaseHttpResponse<actix_web::dev::Body>`
              found fn pointer `fn(&error::CorsError) -> actix_web::HttpResponse`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0053`.
error: could not compile `actix-cors`.

```

## Steps to Reproduce
1) Create a new project with `actix-cors = "0.6.0-beta.1"` as dependency:
```toml
[package]
name = "scratch"
version = "0.1.0"
edition = "2018"

[dependencies]
actix-cors = "0.6.0-beta.1"
```

2) Compile the project with `cargo build`

## Environment

* Rust Version: `rustc 1.53.0-nightly`
* Actix-* crate(s) Version: `actix-cors 0.6.0-beta.1`

**Comment by nhruo123** [comment]
It seems like all the create in this repo are failing to build against `actix-cors 0.6.0-beta.1` ( the build fails against `actix-cors 0.5.0-beta.1` as well because cargo uses `0.6.0` instead, so just cloning this repo and building fails ).

The build seems to fail because of 4 main reasons:

1.  `error_response` function on `ResponseError` trait return type was changed from `HttpResponse` to `BaseHttpResponse<actix_web::dev::Body>`
2. `CookieJar` added a new x_mut functions to get a mutable access over cookies
3. `ServiceRequest` lost his `cookies` function
4. `serde::Error` need to be casted into the new `JsonPayloadError`

I have very little experience with this project so take all the above with a grain of salt.

**Comment by nhruo123** [comment]
I tried to fix it myself, right now everything compiles besides the `actix-identity` create, the `session` create tests and the `identity` create tests. if it can be of any help here is a [link](https://github.com/nhruo123/actix-extras/tree/beta-6-patch) to my changes.

If you don't mind downgrading to beta.5 you can use [this](https://github.com/actix/actix-web/issues/2185#issuecomment-824266090) workaround for now.

**Comment by miraclx** [comment]
Got it compiling successfully on [my fork](https://github.com/near/actix-extras/tree/actix-web-4-beta.6) (https://github.com/actix/actix-extras/compare/master...near:actix-web-4-beta.6).
After updating everything, I added a helper crate that extends `actix_http::HttpMessage` to reinclude the `cookie` and `cookies` methods on `ServiceRequest`.

So at this point, everything compiles.

I won't be creating a PR because this introduces an entirely new crate into the repo which won't be necessary after #2177's merged.

**Comment by aliemjay** [comment]
This should be fixed by #188
